diff --git a/build.rs b/build.rs
index d14c2dd..1542a6b 100644
--- a/build.rs
+++ b/build.rs
@@ -53,6 +53,12 @@ fn main() {
     // libc FFI calls, so if we're running under miri, use the libc backend.
     let miri = var("CARGO_CFG_MIRI").is_ok();
 
+     // skip rustix statx_flags mod.
+     let musl_geq_1_2_3 = var("RUST_LIBC_MUSL_GEQ_1_2_3").is_ok();
+
+    // skip rustix statx_flags mod.
+    let musl_geq_1_2_3 = var("RUST_LIBC_MUSL_GEQ_1_2_3").is_ok();
+
     // If experimental features are enabled, auto-detect and use available
     // features.
     if rustc_dep_of_std {
@@ -159,6 +165,16 @@ fn main() {
         use_feature("fix_y2038");
     }
 
+    // Feature needed for building.
+     if musl_geq_1_2_3 {
+         use_feature("musl_geq_1_2_3");
+     }
+
+    // Feature needed for building.
+    if musl_geq_1_2_3 {
+        use_feature("musl_geq_1_2_3");
+    }
+
     println!("cargo:rerun-if-env-changed=CARGO_CFG_RUSTIX_USE_EXPERIMENTAL_ASM");
     println!("cargo:rerun-if-env-changed=CARGO_CFG_RUSTIX_USE_LIBC");
 
@@ -167,6 +183,12 @@ fn main() {
     println!("cargo:rerun-if-env-changed=CARGO_FEATURE_USE_LIBC");
     println!("cargo:rerun-if-env-changed=CARGO_FEATURE_RUSTC_DEP_OF_STD");
     println!("cargo:rerun-if-env-changed=CARGO_CFG_MIRI");
+
+    println!("cargo:rustc-check-cfg=cfg(musl_geq_1_2_3)");
+    println!("cargo:rerun-if-env-changed=RUST_LIBC_MUSL_GEQ_1_2_3");
+
+    println!("cargo:rustc-check-cfg=cfg(musl_geq_1_2_3)");
+    println!("cargo:rerun-if-env-changed=RUST_LIBC_MUSL_GEQ_1_2_3");
 }
 
 fn use_static_assertions() -> bool {
